controllers/articles.controller.js

```diff
@@ -1,8 +1,40 @@
-const { selectArticleById } = require("../models/articles.model");
+const {
+  selectAllArticles,
+  selectArticleById,
+} = require("../models/articles.model");
+const { countCommentsByArticleID } = require("../models/comments.model");
+
+const getAllArticles = (request, response, next) => {
+  selectAllArticles()
+    .then((articles) => {
+      const commentsPromises = articles.map((article) => {
```

Comment:
This really should be a psql query to add this column rather than altering the data after the fact. You can do all of this in one pqsl query without a promise.all